packages/components/src/components/lifecycleValidation.spec.ts

```diff
@@ -136,6 +148,20 @@ it.each<TagName>(tagNamesPublicWithoutProps)('should not call validateProps() fo
   expect(spy).not.toBeCalled();
 });
 
+it.each<TagName>(tagNamesWithPropsOfTypeObject)(
+  'should call isDeepEqual() with correct parameters for %s',
+  (tagName) => {
+    const spy = jest.spyOn(isDeepEqualUtils, 'isDeepEqual');
+    const component = componentFactory(tagName);
+
+    if (component.componentShouldUpdate) {
```

Comment:
What is the check for? That lifecycle should always be defined now since it is part of the rule.